kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -140,6 +140,8 @@ private CompletableFuture<ProduceResponse> produce(
       throw new StacklessCompletionException(e);
     }
 
+    recordRequestMetrics(request.getOriginalSize());
```

Comment:
nit: I think this deserves a comment explaining the importance of its placement here - that it needs to happen after rate-limiting but before request validation and serialization.